src/jaxsim/api/kin_dyn_parameters.py

```diff
@@ -382,21 +382,20 @@ def joint_transforms_and_motion_subspaces(
         )
 
         # Compute the transforms and motion subspaces of the joints.
-        # TODO: understand how to use joint_indices to access joint_types, right now
-        #       it fails when used within a JIT context.
-        pre_H_suc_and_S = [
-            supported_joint_motion(
-                joint_type=self.joint_model.joint_types[i + 1],
-                joint_position=jnp.array(s),
+        if self.number_of_joints() == 0:
+            pre_H_suc_J, S_J = jnp.empty((0, 4, 4)), jnp.empty((0, 6, 1))
+        else:
+            pre_H_suc_J, S_J = jax.vmap(supported_joint_motion)(
+                jnp.array(self.joint_model.joint_types[1:]).astype(int),
+                jnp.array(self.joint_model.joint_axis),
+                jnp.array(joint_positions),
             )
-            for i, s in enumerate(jnp.array(joint_positions).astype(float))
-        ]
 
         # Extract the transforms and motion subspaces of the joints.
         # We stack the base transform W_H_B at index 0, and a dummy motion subspace
         # for either the fixed or free-floating joint connecting the world to the base.
-        pre_H_suc = jnp.stack([W_H_B] + [H for H, _ in pre_H_suc_and_S])
-        S = jnp.stack([jnp.vstack(jnp.zeros(6))] + [S for _, S in pre_H_suc_and_S])
+        pre_H_suc = jnp.vstack([W_H_B[None, ...], pre_H_suc_J])
+        S = jnp.vstack([jnp.zeros((6, 1))[None, ...], S_J])
```

Comment:
```suggestion
        pre_H_suc = jnp.vstack([W_H_B[jnp.newaxis, ...], pre_H_suc_J])
        S = jnp.vstack([jnp.zeros((6, 1))[jnp.newaxis, ...], S_J])
```

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/153/commits/1b37776144cbc499954751943afe80a2ffb46dbb